app/services/search/base_service.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Search
+  class BaseService
+    def initialize(search_url)
+      @search_url = search_url
+    end
+
+    def call
+      uri = URI.parse(search_url)
+
+      response = Net::HTTP.get_response(uri)
```

Comment:
It might be better to use Faraday or Httparty as this standard library is quite poor (especially to handle errors and using different user agents, the latter will be needed)